Projects/Phase14-Ontology/Phase14-Ontology.md

```diff
@@ -0,0 +1,190 @@
+آنتولوژی از این هم فراتر می‌رود و کمک می‌کند دو دیتابیسی  را که با هم متفاوت هستند به هم مرتبط کنیم. چون همه‌ قواعد دیدگاه دو دیتابیس مشخص شده‌ و می‌توانند به یکدیگر ترجمه‌شوند.
+<br>
+یک قبیله دورافتاده در یک جنگل را فرض‌کنید. این قبیله نمی‌تواند با مردم نقاط دیگر جهان ارتباط بگیرد. هم‌چنین ارتباط افراد قبیله بایکدیگر هم به سختی صورت می‌گیرد. بعیدمی‌دانم این قبیله در این دنیا چیز خیلی جذابی به نظر برسد.
+<br>
+سیستمی که آنتولوژیش درست تعریف نشده‌باشد دقیقا شبیه همین قبیله است.
+
+## ۳- آنتولوژی پویا چیست و چرا به آن نیاز داریم؟
+بالاتر که در مورد آنتولوژی صحبت کردیم، توضیح دادیم که مثلا یک‌‌ شماره‌حساب را می‌توانیم نود گراف ببینیم و یا این‌که طور دیگری نگاه‌کنیم و آن را یک لینک بدانیم. مهم‌تر این که هر کدام از این دیدگاه‌ها محاسن منحصربه‌فردی دارند.
+<br>
+تا مدت‌ها وقتی پلت‌فرمی را توسعه می‌دادند همان ابتدا آنتولوژی را ست‌می‌‌کردند و تغییر آن نیاز به ساخت محصول جدیدداشت. یعنی آنتولوژی آن‌ها ثابت بود. اما امروزه دنیا برای جلوگیری از دوباره‌کاری و بهره‌مندشدن از امکانات آنتولوژی‌های مختلف، به سمتی رفته که امکان تغییر آنتولوژی را در همان محصول به مشتری بدهد. یعنی محصول را طوری بسازیم که با گرفتن اطلاعات از ادمین، سیستم آنتولوژی را تغییر دهد.
+
+## ۴- امکان تعریف آنتولوژی توسط ادمین در فایل
+قبل از توضیح این قسمت، بیایید تا معنی دو کلمه زیر را با یکدیگر قرارداد کنیم.
+<br><br>
+<b>موجودیت</b>: به هر یک از عناصر اصلی تشکیل دهنده سیستمی که برایش پلت‌فرم را توسعه می‌دهیم، موجودیت می‌گویند. مثل شخص، تراکنش، بانک و ... . این کلمه معادل عبارت Entity است.
+<br>
+<b>هویت</b>: به جنس هر کدام یک از موجودیت‌ها در این سیستم هویت گفته می‌شود. در سیستم ما منظور از هویت، لینک بودن، نودبودن و مشخصه بودن است. مثلا اگر کاربر تصمیم‌گرفت به یک شخص به عنوان نود گراف نگاه کند یعنی هویت شخص در سیستم ما نودگراف است.
+<br><br>
+برای تعیین آنتولوژی توسط کاربر، او باید یک فایل جیسون را در سیستم بارگزاری کند. در این فایل موارد زیر مشخص می‌شود.
+<br>
+*  چه چیزی نود است
+* چه چیزی لینک است
+*  و چه چیزی مشخصه اجزای تشکیل دهنده گراف است.بعد از این‌که نودها و لینک‌های گراف ما مشخص شد و موجودیت‌ها با هویت جدید شناخته‌شدند، مشخصه‌های اجزای گراف و تایپ آن مشخصه‌ها  معلوم می‌شود. در اینجا منظور از مشخصه همان attribute و منظور از تایپ،string، integer و ... است.
+<br>
+
+برای مثال، ادمین مشخص می‌کند که تراکنش باید لینک گراف، شخص انجام‌دهنده تراکنش به عنوان نود گراف و تاریخ انجام تراکنش به عنوان مشخصه آن تراکنش باشد.در این زیر شما یک نمونه فایل جیسون را که برای تعیین آنتولوژی به کار می‌رود مشاهده می‌کنید.
+
+<div dir="ltr">
+  
+    {
+        "nodes" :{
+            "Person":{
+                "Name": "string",
+                "FamilyName": "string",
+                "NationalCod" : "string",
+                "Birthday" : "date"   
+            },
+        
+            "Bank":{
+                "Name": "string",
+                "BranchID": "string",
+                "Adress": "string"
+            }
+
+            "Account":{
+                "AccountNum": "string"
+                "OpenningDate": "date"
+            }
+        },
+
+        "links":{
+            "Transaction":{
```

Comment:
Each link should have SourceNodeType & TargetNodeType, for example for transaction SourceNodeType is Account and TargetNodeType is Account too.